tool/tbot/config/configtemplate.go

```diff
@@ -88,7 +88,7 @@ type Template interface {
 	// statically as this must be callable without any auth clients (or any
 	// secrets) for use with `tbot init`. If an arbitrary number of files must
 	// be generated, they should be placed in a subdirectory.
-	Describe() []FileDescription
+	Describe(destination destination.Destination) []FileDescription
```

Comment:
It's come up before and I agree that it's ugly, but this practically needs to live in its own package to avoid import cycles. Since it's the only thing that will ever live in that package, there's few names that would make sense other than, well, `destination`. Always open to suggestions if you've got a more creative solution to offer :slightly_smiling_face:

Reply:
Having a quick look the only implementor I found was DestinationDirectory. The `destination` package holds an interface and nothing else, which is a bit of a design smell for Go.

I would suggest the following:

* Rename `package destination` to `package destinations`. The main reason is that we can have variables named `destination` without overshadowing the package name (similar to `strings` or `bytes`).
* Move `DestinationDirectory` to `destinations.Directory`. Names fit nicely and the package now exports something concrete. (IIUC, this won't cause import cycles.)
* Ditch the interface, or at the very least ditch it from the `destinations` package. If you do need an interface, spring up one where it's used, preferably with only 1 or 2 methods that the code cares about.

We don't usually need "preemptive" interfaces in Go. Code against the concrete type if you can; if you need an interface, spit it out where it's needed, and only after it has a reason to be there.  (The api/ package is the one place where we may want to do differently, but for internal code this should work just fine.)